packages/vitest-pool-workers/src/worker/fetch-mock.ts

```diff
@@ -21,17 +21,19 @@ interface BufferedRequest {
 	request: Request;
 	body: Uint8Array | null;
 }
-const requests = new WeakMap<Dispatcher.DispatchOptions, BufferedRequest>();
-const responses = new WeakMap<Dispatcher.DispatchOptions, Response>();
+const requests = new Map<string, BufferedRequest>();
+const responses = new Map<string, Response>();
```

Comment:
Previously, we relied on stability of the `DispatchOptions` variable, but that is no longer guaranteed in Undici v7 (see https://github.com/nodejs/undici/blob/c6a8b23ee55b34fedd7b90d3c7f4b51a35d3ee25/lib/mock/mock-agent.js#L74). As such, we serialise the options for lookup, and make sure we remove them manually from the `Map` after access so as not to cause a memory leak.